Source/Evolve/WarpXEvolve.cpp

```diff
@@ -136,7 +136,32 @@ WarpX::Evolve (int numsteps)
                             *Efield_aux[lev][0],*Efield_aux[lev][1],*Efield_aux[lev][2],
                             *Bfield_aux[lev][0],*Bfield_aux[lev][1],*Bfield_aux[lev][2]);
             }
+
+            // The hybrid-PIC algorithm uses the charge and current density
+            // from both the current and previous step when updating the
+            // fields, so we deposit the ion charge and current now in the
+            // temp multifab locations.
+            if (electromagnetic_solver_id == ElectromagneticSolverAlgo::HybridPIC)
+            {
+                mypc->DepositCharge(rho_fp_temp, 0._rt);
+                mypc->DepositCurrent(current_fp_temp, dt[0], 0._rt);
+                SyncCurrent(current_fp_temp, current_cp);
```

Comment:
Is there a reason to not use `SyncRho` here instead of calling `AddRhoFromFineLevelandSumBoundary` directly?

Also, perhaps add new routines like `void WarpX::ApplyRhofieldBoundary (PatchType patch_type)` that have the loop over levels, as is done elsewhere.

Reply:
If there was a reason for not calling `SyncRho` directly, I don't remember what it was - maybe I only saw the `SyncRho()` function which always operates on `rho_fp` and not the version below it that takes multifabs as input - but I will change that to be more consistent with the current density treatment. Thanks for pointing it out!

I think implementing `ApplyRhofieldBoundary (PatchType patch_type)` functions that will include the loop over levels can maybe be done in a separate PR since for consistency it should be done for E, B, J and rho.

Reply:
I add these routines for E and B in PR #3842. This PR can add them for J and rho. But a separate PR is fine also.

Reply:
Sounds good. I'll keep an eye on your PR and submit a follow-up for J and rho once it is merged.